source/routing/raptor.cpp

```diff
@@ -81,9 +81,9 @@ bool RAPTOR::apply_vj_extension(const Visitor& v, const bool global_pruning,
             const auto sp_idx = SpIdx(*sp);
             const DateTime bound = (v.comp(best_labels_pts[sp_idx], b_dest.best_now) || !global_pruning) ?
                                         best_labels_pts[sp_idx] : b_dest.best_now;
-            if(!v.comp(workingDt, bound)) {
-                continue;
-            }
+            if (v.comp(bound, workingDt)) { continue; }
+            if (!v.comp(workingDt, working_labels.dt_pt(sp_idx))) { continue; }
```

Comment:
I don't really know how to do that cleanly as it is done in several different location, here and in apply_vj_extention, and the equivalent is done in foot_path. Adding comments at all these locations seems dangerous, but I do not have any better idea. Do you have?